onnxruntime/core/platform/windows/dll_load_error.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <windows.h>
+#include <dbghelp.h>
+#include <iostream>
+#include <string>
+#pragma comment(lib, "dbghelp.lib")
+
+struct HMODULE_Deleter {
+  typedef HMODULE pointer;
+  void operator()(HMODULE h) { FreeLibrary(h); }
+};
+
+using ModulePtr = std::unique_ptr<HMODULE, HMODULE_Deleter>;
+
+// If a DLL fails to load, this will try loading the DLL and then its dependencies recursively
+// until it finds a missing file, then will report which file is missing and what the dependency
+// chain is.
+std::wstring DetermineLoadLibraryError(const wchar_t* filename_in) {
+  std::wstring error(L"Error loading");
+
+  std::wstring filename{filename_in};
+  while (filename.size()) {
+    error += std::wstring(L" \"") + filename + L"\"";
+
+    ModulePtr hModule = ModulePtr{LoadLibraryExW(filename.c_str(), NULL, DONT_RESOLVE_DLL_REFERENCES)};
```

Comment:
use LOAD_LIBRARY_AS_DATAFILE here?

Reply:
I tried it but it doesn't work, it results in all of the dependent dll names being nullptr. The OS must need to do some processing of the imports for this table to be valid.